cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go

```diff
@@ -371,3 +394,533 @@ func TestMachinesInMachineSet(t *testing.T) {
 		}
 	}
 }
+
+func TestControllerNodeGroupsSizes(t *testing.T) {
+	for i, tc := range []struct {
+		description string
+		annotations map[string]string
+		count       int
+	}{{
+		description: "errors because minSize is invalid",
+		annotations: map[string]string{
+			nodeGroupMinSizeAnnotationKey: "-1",
+			nodeGroupMaxSizeAnnotationKey: "0",
+		},
+	}, {
+		description: "errors because maxSize is invalid",
+		annotations: map[string]string{
+			nodeGroupMinSizeAnnotationKey: "0",
+			nodeGroupMaxSizeAnnotationKey: "-1",
+		},
+	}, {
+		description: "errors because minSize > maxSize",
+		annotations: map[string]string{
+			nodeGroupMinSizeAnnotationKey: "1",
+			nodeGroupMaxSizeAnnotationKey: "0",
+		},
+	}, {
+		description: "errors because maxSize < minSize",
+		annotations: map[string]string{
+			nodeGroupMinSizeAnnotationKey: "1",
+			nodeGroupMaxSizeAnnotationKey: "0",
+		},
+	}, {
+		annotations: map[string]string{
+			nodeGroupMaxSizeAnnotationKey: "10",
+		},
+		count: 1,
+	}, {
+		annotations: map[string]string{
+			nodeGroupMinSizeAnnotationKey: "1",
+			nodeGroupMaxSizeAnnotationKey: "10",
+		},
+		count: 1,
+	}} {
+		machineSet := &v1beta1.MachineSet{
+			TypeMeta: v1.TypeMeta{
+				Kind: "MachineSet",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:        fmt.Sprintf("machineset-%d", i),
+				Namespace:   "test-namespace",
+				Annotations: tc.annotations,
+			},
+		}
+
+		controller, stop := mustCreateTestController(t, testControllerConfig{
+			machineObjects: []runtime.Object{
+				machineSet,
+			},
+		})
+		defer stop()
+
+		nodegroups, err := controller.nodeGroups()
+		if tc.count == 0 && err == nil {
+			t.Fatalf("expected an error")
+		}
+
+		if l := len(nodegroups); l != tc.count {
+			t.Errorf("expected %v, got %v", tc.count, l)
+		}
+	}
+}
+
+func TestControllerNodeGroupForNodeWithMissingNode(t *testing.T) {
```

Comment:
For node with missing node? Can you more clarify on what is purpose of the test?

Reply:
Addressed in https://github.com/openshift/kubernetes-autoscaler/pull/34/commits/5b7727ae59eef49d6ab4d9cb64efac909b60d347